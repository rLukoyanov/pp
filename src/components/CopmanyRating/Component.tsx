import styles from './styles.module.scss';

export function CompanyRating() {
  return (
    <div className={styles.rating}>
      <div className={styles.groups}>
        <div>Место</div>
        <div>Компания</div>
        <div>Отрасль</div>
        <div className={styles.students}>Принято студентов</div>
      </div>
      <div className={styles['rating-item']}>
        <div>1</div>
        <span>iPlast</span>
        <div>Промышленность</div>
        <div className={styles.students}>30 студентов</div>
      </div>
      <div className={styles['rating-item']}>
        <div>2</div>
        <span>Татнефть</span>
        <div>Промышленность</div>
        <div className={styles.students}>30 студентов</div>
      </div>
      <div className={styles['rating-item']}>
        <div>3</div>
        <span>СДЭК</span>
        <div>Логистика</div>
        <div className={styles.students}>30 студентов</div>
      </div>
      <div className={styles['rating-item']}>
        <div>4</div>
        <span>iPlast</span>
        <div>Промышленность</div>
        <div className={styles.students}>30 студентов</div>
      </div>
    </div>
  );
}
