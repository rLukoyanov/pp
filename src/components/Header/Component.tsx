import cn from 'classnames';

import styles from './styles.module.scss';

export function Header() {
  const classNames = cn(styles.header);
  return (
    <div className={classNames}>
      <div>search</div>
      <div>logo</div>
    </div>
  );
}
