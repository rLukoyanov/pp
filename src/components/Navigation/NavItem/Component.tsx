import cn from 'classnames';
import { Link } from 'react-router-dom';

import styles from './styles.module.scss';

export function NavItem({
  children = '',
  active = false,
  icon = '',
  url = '/',
}) {
  const classNames = cn(styles['nav-item'], { [styles.active]: active });

  const changeActive = () => {};

  return (
    <Link style={{ textDecoration: 'none', color: 'black' }} to={url}>
      <div className={classNames}>
        <img src={icon} />
        <span onClick={changeActive}>{children}</span>
      </div>
    </Link>
  );
}
