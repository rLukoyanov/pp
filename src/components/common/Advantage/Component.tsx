import classNames from 'classnames';
import { ReactNode } from 'react';
import styles from './styles.module.scss';

export function Advantage({
  children,
  last = false,
  top = false,
  left = false,
  color = false,
}: {
  children: ReactNode;
  last?: boolean;
  top?: boolean;
  left?: boolean;
  color?: boolean;
}) {
  return (
    <div
      className={classNames(styles.advantage, {
        [styles.last]: last,
        [styles.top]: top,
        [styles.left]: left,
        [styles.color]: color,
      })}
    >
      {children}
    </div>
  );
}
