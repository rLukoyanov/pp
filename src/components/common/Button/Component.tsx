import cn from 'classnames';

import styles from './styles.module.scss';

export function Button({
  children = '',
  className = '',
  disabled = false,
  type = '',
}) {
  const classNames = cn(styles.button, {
    [styles.outline]: type === 'outline',
  });

  return (
    <button className={`${classNames} ${className}`} disabled={disabled}>
      {children}
    </button>
  );
}
