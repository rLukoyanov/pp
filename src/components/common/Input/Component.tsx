import { useState } from 'react';

import classNames from 'classnames';

import InputArrow from '../../../images/input/inputArrow.svg';
import InputDownArrow from '../../../images/input/inputDownArrow.svg';

import styles from './styles.module.scss';

export function Input({
  status = 'default',
  disabled = false,
  leftIcon = false,
  rightIcon = false,
  hint = false,
  label = '',
  placeholder = '',
}) {
  const [focused, setFocused] = useState(false);
  const onFocus = () => setFocused(true);
  const onBlur = () => setFocused(false);
  const changeStyles = classNames(styles.outline, {
    [styles.error]: status === 'error',
    [styles.success]: status === 'success',
    [styles.error]: status === 'error',
  });

  const inputStyles = classNames(styles.input, {
    [styles.left]: leftIcon,
    [styles.right]: rightIcon,
  });

  return (
    <div className={styles['input-container']}>
      <div className={styles.label}>{label}</div>
      <div
        className={
          focused
            ? `${styles.outline} ${styles.focused}`
            : `${styles.outline} ${changeStyles}`
        }
      >
        {leftIcon ? <img alt="inputArrow" src={InputArrow} /> : <></>}
        <input
          onFocus={onFocus}
          onBlur={onBlur}
          className={inputStyles}
          placeholder={placeholder}
          type="text"
          disabled={disabled}
        />
        {rightIcon ? <img alt="inputDownArrow" src={InputDownArrow} /> : <></>}
      </div>
      {hint ?? <div className={styles.hint}>hint text</div>}
    </div>
  );
}
