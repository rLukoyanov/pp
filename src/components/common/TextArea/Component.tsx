import styles from './styles.module.scss';

import classNames from 'classnames';

export function TextArea({ status = 'default', disabled = false }) {
  const inputStyles = classNames(styles.textarea, {
    [styles.outline]: status === 'default',
    [styles.error]: status === 'error',
    [styles.success]: status === 'success',
    [styles.disabled]: status === 'disabled',
  });

  return (
    <textarea
      className={inputStyles}
      placeholder="placeholder"
      disabled={disabled}
    />
  );
}
