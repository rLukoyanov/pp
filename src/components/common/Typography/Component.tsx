import styles from './styles.module.scss';

export function Input({ href = '', children = <></> }) {
  return (
    <a href={href} className={styles.link}>
      {children}
    </a>
  );
}
