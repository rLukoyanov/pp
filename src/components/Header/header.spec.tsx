import { render, screen } from '@testing-library/react';
import { Header } from './Component';

test('renders learn react link', () => {
  render(<Header />);
  const headerElement = screen.getByPlaceholderText(/Search/i);

});
