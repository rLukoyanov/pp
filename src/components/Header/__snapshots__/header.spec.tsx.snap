// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<input
  class="input left"
  placeholder="Search"
  type="text"
/>
`;
